apps/silero-vad-whisper-realtime-api/src/router.rs

```diff
@@ -0,0 +1,393 @@
+use std::{collections::HashMap, sync::Arc, time::Instant};
+
+use anyhow::Result;
+use axum::{
+  Json,
+  extract::{Multipart, State},
+  http::StatusCode,
+  response::{
+    IntoResponse,
+    Response,
+    sse::{Event, KeepAlive, Sse},
+  },
+};
+use futures::stream::{self, Stream};
+use symphonia::{
+  core::{
+    audio::{AudioBufferRef, Signal},
+    codecs::DecoderOptions,
+    formats::FormatOptions,
+    io::MediaSourceStream,
+    meta::MetadataOptions,
+    probe::Hint,
+  },
+  default::get_probe,
+};
+
+use crate::{
+  AppState,
+  api::{ErrorDetail, ErrorResponse, StreamChunk, TranscriptionResponse},
+  audio_manager::AudioBuffer,
+};
+
+// Performance statistics struct
+#[derive(Debug)]
+struct ProcessingStats {
+  total_duration:                 std::time::Duration,
```

Comment:
```suggestion
  total_duration:                 std::time::Duration,
```

If imported multiple times, use them all at the top.